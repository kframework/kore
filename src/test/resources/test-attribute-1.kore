[]

module TEST-ATTRIBUTE-1

  sort Nat{} [hook{}(\dv{Hook{}}("Nat"))]

endmodule []
