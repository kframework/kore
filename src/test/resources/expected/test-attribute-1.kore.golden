Definition
    { definitionAttributes = Attributes []
    , definitionModules =
        [ Module
            { moduleName = ModuleName "TEST-ATTRIBUTE-1"
            , moduleSentences =
                [ ObjectSentence (SentenceSortSentence SentenceSort
                    { sentenceSortName = Id "Nat" :: Id Object
                    , sentenceSortParameters = []
                    , sentenceSortAttributes =
                        Attributes 
                            [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                { applicationSymbolOrAlias =
                                    SymbolOrAlias
                                        { symbolOrAliasConstructor = Id "hook" :: Id Object
                                        , symbolOrAliasParams = []
                                        }
                                , applicationChildren =
                                    [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (DomainValuePattern DomainValue
                                        { domainValueSort =
                                            SortActualSort SortActual
                                                { sortActualName = Id "Hook" :: Id Object
                                                , sortActualSorts = []
                                                }
                                        , domainValueChild =
                                            Fix (UnifiedPattern (UnifiedMeta (Rotate31 (StringLiteralPattern (StringLiteral "Nat")))))
                                        }))))
                                    ]
                                }))))
                            ]
                    })
                ]
            , moduleAttributes = Attributes []
            }
        ]
    }